python/delta_sharing/rest_client.py

```diff
@@ -64,15 +67,37 @@ class ListFilesInTableResponse:
     add_files: Sequence[AddFile]
 
 
+def retry_with_exponential_backoff(func):
+    def func_with_retry(self, *arg, **kwargs):
+        times_retried = 0
+        sleep_ms = 100
+        while True:
+            times_retried += 1
+            try:
+                return func(self, *arg, **kwargs)
+            except Exception as e:
+                if self._should_retry(e) and times_retried <= self._num_retries:
+                    logging.info(f"Sleeping {sleep_ms} to retry because of error {e}")
+                    self._sleeper(sleep_ms)
+                    sleep_ms *= 2
+                else:
+                    raise e
+
+    return func_with_retry
+
```

Comment:
Had to put this outside because I get errors if I do `retry_with_exponential_backoff(self, func)`
https://stackoverflow.com/questions/38524332/declaring-decorator-inside-a-class